### PR TITLE
Mint with permission

### PR DESCRIPTION
**Pull request by rahulrumalla**
Fixes https://github.com/rugpullindex/ERC4973/issues/16 and https://github.com/rugpullindex/ERC4973/issues/17. Follows up on the work for https://github.com/rugpullindex/ERC4973/issues/7

### CODE REVIEW

**Comment by rahulrumalla** [comment]
I went back and forth on this very idea u r suggesting.

One issue is that if `getMintPermitMessageHash` is part of the interface, then it needs to be marked as `external`. Meaning that `_isPermittedToMint` now cannot call the function anymore due to visibility issues.

I think since _signTypedData will produce the same result, we should be fine here not exposing it in interface.

File: src/interfaces/IERC4973Permit.sol [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.6;
+
+/// @title Account-bound tokens, optional permissioned minting extension
+/// @dev See https://eips.ethereum.org/EIPS/eip-4973
```
Comment (TimDaub) [inline src/interfaces/IERC4973Permit.sol:5]: please add the function identifier as a hex value here as in the other interface: https://github.com/rugpullindex/ERC4973/pull/18/files#diff-d6d069dfcb878b5b981b7d3dd18dce17746639eea3f30d1902f4a4cd46364f84R6

You can generate the interface identifier with a similar test as this one: https://github.com/rugpullindex/ERC4973/blob/9c6bd57c4f34da5a1507ec2ff6cb3b105594d712/src/ERC4973.t.sol#L50

**Review by TimDaub — CHANGES_REQUESTED** [review]
only requested change is to add the interface id as hex value as comment. Otherwise great job!

**Comment by TimDaub** [comment]
> I think since _signTypedData will produce the same result, we should be fine here not exposing it in interface.

OK, makes sense.

**Comment by rahulrumalla** [comment]
Would we need to add a `supportsInterface` override in `ERC4973Permit.sol` ?

**Comment by TimDaub** [comment]
> Would we need to add a `supportsInterface` override in `ERC4973Permit.sol` ?

yes probably. May be worth writing a test and checking all identifiers that are supposed to yield `true` for `ERC4973Permit`

File: src/ERC4973Permit.sol [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 import {EIP712} from "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
-
 import {ERC4973} from "./ERC4973.sol";
+import {IERC4973Permit} from "./interfaces/IERC4973Permit.sol"; 
 
 /// @notice Reference implementation of ERC4973Permit
-/// @author Rahul Rumalla, Tim Daub
-abstract contract ERC4973Permit is ERC4973, EIP712 {
+/// @author Rahul Rumalla, Tim Daub (https://github.com/rugpullindex/ERC4973/blob/master/src/ERC4973Permit.sol)
```
Comment (TimDaub) [inline src/ERC4973Permit.sol:10]:
```suggestion
/// @author Rahul Rumalla, Tim Daubenschuetz (https://github.com/rugpullindex/ERC4973/blob/master/src/ERC4973Permit.sol)
```

**Review by TimDaub — APPROVED** [review]
LGTM 🎊

**Pull request merged**